src/core/server/plugins/plugins_system.ts

```diff
@@ -159,7 +159,15 @@ export class PluginsSystem {
       const pluginName = this.satupPlugins.pop()!;
 
       this.log.debug(`Stopping plugin "${pluginName}"...`);
-      await this.plugins.get(pluginName)!.stop();
+
+      const result = (await Promise.race([
+        this.plugins.get(pluginName)!.stop(),
+        new Promise((resolve) => setTimeout(() => resolve({ delayed: true }), 30 * Sec)),
```

Comment:
With https://github.com/elastic/kibana/pull/89562 (https://github.com/elastic/kibana/issues/71925), we changed to `setup` and `start` timeout to `10s`. We might want to have the same value for `stop`?

Reply:
Since async stop isn't deprecated and we want to give plugins a chance to do a graceful shutdown (https://github.com/elastic/kibana/issues/83612) 10s feels too short.

Reply:
SGTM. Do we want to change the `Plugin` and `AsyncPlugin`'s `stop` signature to reflect that returning a Promise is supported? atm it's `stop?(): void;`

Reply:
@pgayvallet I'd wait for an official decision on that matter in https://github.com/elastic/kibana/issues/83612